File: leetcode-cn/CrackingTheCodingInterview/16.02.0_Words_Frequency_LCCI.py
'''
Hash Table

执行用时：280 ms, 在所有 Python3 提交中击败了63.30% 的用户
内存消耗：43.4 MB, 在所有 Python3 提交中击败了78.11% 的用户
通过测试用例：24 / 24
'''
class WordsFrequency:

    def __init__(self, book: List[str]):
        self.c = Counter(book)

    def get(self, word: str) -> int:
        return self.c[word]

# Your WordsFrequency object will be instantiated and called as such:
# obj = WordsFrequency(book)
# param_1 = obj.get(word)

File: js/16.02.0_Words_Frequency_LCCI.js
/*
执行用时：368 ms, 在所有 JavaScript 提交中击败了87.50% 的用户
内存消耗：69.4 MB, 在所有 JavaScript 提交中击败了56.82% 的用户
通过测试用例：24 / 24
*/

/**
 * @param {string[]} book
 */
var WordsFrequency = function(book) {
    this.map = {}
    for (let item of book) {
        if (!this.map[item]) {
            this.map[item] = 1
        } else {
            this.map[item]++
        }
    }
};

/** 
 * @param {string} word
 * @return {number}
 */
WordsFrequency.prototype.get = function(word) {
    return this.map[word] ? this.map[word] : 0
};

/**
 * Your WordsFrequency object will be instantiated and called as such:
 * var obj = new WordsFrequency(book)
 * var param_1 = obj.get(word)
 */
 